showcase/types/global.d.ts

```diff
@@ -2,138 +2,29 @@
  * SPDX-License-Identifier: MPL-2.0
  */
-
-/* eslint-disable ember/require-tagless-components */
-/* eslint-disable @typescript-eslint/ban-types */
-
 import '@glint/environment-ember-loose';
-import Component from '@glimmer/component';
 
 import type PageTitle from 'ember-page-title/template-registry';
+import type EmberTruthRegistry from 'ember-truth-helpers/template-registry';
+import type RenderModifiersRegistry from '@ember/render-modifiers/template-registry';
 
-export interface ShwTxtSignature {
-  Args: {
-    tag?: string;
-    variant?: string;
-    align?: string;
-    weight?: string;
-  };
-  Element: HTMLElement;
-  Blocks: {
-    default: [];
-  };
-}
-export class ShwTxt extends Component<ShwTxtSignature> {}
-
-export interface ShwLabelSignature {
-  Args: {};
-  Element: HTMLParagraphElement;
-  Blocks: {
-    default: [];
-  };
-}
-export class ShwLabel extends Component<ShwLabelSignature> {}
-
-export interface ShwFlexItemSignature {
-  Args: {
-    grow?: boolean;
-    label?: string;
-  };
-  Element: HTMLDivElement;
-  Blocks: {
-    default: [
-      {
-        Label: typeof ShwLabel;
-      },
-    ];
-  };
-}
-export class ShwFlexItem extends Component<ShwFlexItemSignature> {}
-
-export interface ShwFlexSignature {
-  Args: {
-    direction?: string;
-    wrap?: boolean;
-    label?: string;
-    gap?: string;
-  };
-  Element: HTMLDivElement;
-  Blocks: {
-    default: [
-      {
-        Label: typeof ShwLabel;
-        Item: typeof ShwFlexItem;
-      },
-    ];
-  };
-}
-export class ShwFlex extends Component<ShwFlexSignature> {}
-
-export interface ShwGridItemSignature {
-  Args: {
-    grow?: boolean;
-    forceMinWidth?: boolean;
-    label?: string;
-  };
-  Element: HTMLDivElement;
-  Blocks: {
-    default: [
-      {
-        Label: typeof ShwLabel;
-      },
-    ];
-  };
-}
-export class ShwGridItem extends Component<ShwGridItemSignature> {}
-
-export interface ShwGridSignature {
-  Args: {
-    columns: number;
-    label?: string;
-    gap?: string;
-  };
-  Element: HTMLDivElement;
-  Blocks: {
-    default: [
-      {
-        Label: typeof ShwLabel;
-        Item: typeof ShwGridItem;
-      },
-    ];
-  };
-}
-export class ShwGrid extends Component<ShwGridSignature> {}
-
-export interface ShwDividerSignature {
-  Args: {
-    level?: number;
-  };
-}
-export class ShwDivider extends Component<ShwDividerSignature> {}
-
-export interface ShwOutlinerSignature {
-  Args: {};
-  Element: HTMLDivElement;
-  Blocks: {
-    default: [];
-  };
-}
-export class ShwOutliner extends Component<ShwOutlinerSignature> {}
-
-export interface ShwPlaceholderSignature {
-  Args: {
-    text?: string;
-    width?: string;
-    height?: string;
-    background?: string;
-    flex?: string;
-  };
-  Element: HTMLDivElement;
-  Blocks: {
-    default: [];
-  };
-}
-export class ShwPlaceholder extends Component<ShwPlaceholderSignature> {}
+import ShwAutoscrollable from '../app/components/shw/autoscrollable';
```

Comment:
could we possibly create separate file and import it here as we do with hds and others

Reply:
I can definitely put this in a separate file! And interesting about not needing to add them to the registry, I did notice that when the `shw/label` and `shw/flex/item` weren't included, lines like [this](https://github.com/hashicorp/design-system/blob/01bc8ab3e06744f231d45c0f6726839b630420ab/showcase/app/components/shw/grid/index.hbs#L10) had type errors 

```hbs
  {{yield (hash Label=(component "shw/label"))}}
```

Reply:
Indeed, we likely need them at least for contextual components